ui_components/src/home/Home.js

```diff
@@ -0,0 +1,564 @@
+    +newsApiLink: string,
+    +loginErrorText: string,
+    +registerErrorText: string,
+    +resetErrorText: string,
+    +loginMessageText: string,
+    +registerPreFill: $ReadOnlyArray,
+    +initialNewsPosts: $ReadOnlyArray < NewsPostType >,
+|};
+function Home({
+    newsApiLink,
+    loginErrorText,
+    registerErrorText,
+    resetErrorText,
+    loginMessageText,
+    registerPreFill,
+    initialNewsPosts,
+}: Props) {
+    const [displayLogin, setDisplayLogin] = React.useState((loginErrorText != "" || loginMessageText != "") ? true : false);
+    const [displayRegister, setDisplayRegister] = React.useState(registerErrorText == "" ? false : true);
+    const [displayReset, setDisplayReset] = React.useState(resetErrorText == "" ? false : true);
+    const [displayRules, setDisplayRules] = React.useState(false);
+    const [displayTerms, setDisplayTerms] = React.useState(false);
+    const [newsPosts, setNewsPosts] = React.useState(initialNewsPosts);
+    const newsRef = React.useRef(null);
+    const contactRef = React.useRef(null);
+
+    return (
+        <>
+            <LoginSection
+                displayLogin={displayLogin}
+                setDisplayLogin={setDisplayLogin}
+                displayRegister={displayRegister}
+                setDisplayRegister={setDisplayRegister}
+                displayReset={displayReset}
+                setDisplayReset={setDisplayReset}
+                displayRules={displayRules}
+                setDisplayRules={setDisplayRules}
+                displayTerms={displayTerms}
+                setDisplayTerms={setDisplayTerms}
+                loginErrorText={loginErrorText}
+                registerErrorText={registerErrorText}
+                resetErrorText={resetErrorText}
+                loginMessageText={loginMessageText}
+                registerPreFill={registerPreFill}
+                newsRef={newsRef}
+                contactRef={contactRef}
+            />
+            <NewsSection
+                newsRef={newsRef}
+                newsPosts={newsPosts}
+            />
+            <FeatureSection />
+            <WorldSection />
+            <ContactSection
+                contactRef={contactRef}
+            />
+            <FooterSection />
+        </>
+    );
+}
+
+function LoginSection({
+    displayLogin,
+    setDisplayLogin,
+    displayRegister,
+    setDisplayRegister,
+    displayReset,
+    setDisplayReset,
+    displayRules,
+    setDisplayRules,
+    displayTerms,
+    setDisplayTerms,
+    loginErrorText,
+    registerErrorText,
+    resetErrorText,
+    loginMessageText,
+    registerPreFill,
+    newsRef,
+    contactRef,
+}) {
+    function handleLogin() {
+        setDisplayRegister(false);
+        setDisplayReset(false);
+        setDisplayRules(false);
+        setDisplayTerms(false);
+        if (!displayLogin) {
+            setDisplayLogin(true);
+        }
+        else {
+            document.getElementById('login_form').submit();
+        }
```

Comment:
Looking through these, it doesn't look like there's ever a case we want to have multiple of these be true. What if we moved this to a single `pageToDisplay` state variable, that is a union of `"register" | "reset" | "rules" | "terms" | "login"`

Reply:
Marking as to-do.